asmcli/commands/create-mesh.sh

```diff
@@ -127,6 +127,12 @@ install_one_remote_secret() {
   local CTX_CLUSTER2; CTX_CLUSTER2="${2}"
   local SECRET_NAME; SECRET_NAME="${CTX_CLUSTER1//_/-}"
 
+  if [[ "${#SECRET_NAME}" -gt 63 ]]; then
+    local RAND
+    RAND="$(tr -dc "a-z0-9" </dev/urandom | head -c8 || true)"
+    SECRET_NAME="${SECRET_NAME:0:54}-${RAND}"
+  fi
+
```

Comment:
I think it's probably better here to use something deterministic and not random, to handle the case where the mesh gets re-created with two overlapping but distinct sets of clusters.

Let's assume 50 as a reasonable upper bound on how many clusters will be in a mesh. 20 chars gives a space of 2^20 bits, and that means any two hashes have a ~0.12% chance of colliding, so taking the first 20 chars of a hash and then using the existing secret name feels like a decent tradeoff between human readability and preventing collisions.

You can generate the hash with sha256sum:
```
$ echo "test" | sha256sum
f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2  -
```

Reply:
Done